### PR TITLE
Add API user search endpoint

### PR DESCRIPTION
**Pull request by ashmckenzie**
This PR adds a new `/api/user_search` endpoint that allows one to search for a user given their email address.

/cc @zendesk/samson @grosser 

### Tasks
 - [ ] :+1: from team

### Risks
- Level: Low

### CODE REVIEW

File: app/controllers/api/users_controller.rb [hunk 1]
```diff
@@ -3,7 +3,13 @@ class Api::UsersController < Api::BaseController
   before_action :authorize_resource!
 
   def destroy
-    User.find(params.require(:id)).soft_delete!
```
Comment (grosser) [inline app/controllers/api/users_controller.rb]: old code already returned 404 when not found ... via `.find` failure

**Comment by grosser** [comment]
search endpoint is `index` for most controllers ... how about adding a email filter there ?

File: app/controllers/api/users_controller.rb (hunk 1 above)
Comment (ashmckenzie) [inline app/controllers/api/users_controller.rb]: it also returns a boatload of HTML junk as well :)  this address that

**Comment by ashmckenzie** [comment]
will change to `#index` and user query string

**Comment by ashmckenzie** [comment]
green :)

File: app/controllers/api/users_controller.rb (hunk 1 above)
Comment (grosser) [inline app/controllers/api/users_controller.rb]: then the solution should not be a 1-off fix ... if api 404 renders as html then we need a general fix ...

**Comment by grosser** [comment]
... I'll make a PR to fix the not-found and add the search ... brb

**Comment by grosser** [comment]
404 works just fine :)
https://github.com/zendesk/samson/pull/2160

**Comment by grosser** [comment]
... and here is the search by email api https://github.com/zendesk/samson/pull/2161

**Comment by ashmckenzie** [comment]
Closing in favour of https://github.com/zendesk/samson/pull/2162

**Pull request closed without merging**